### PR TITLE
fix #39

### LINKED ISSUE #39: Make the chart clickable
**Current behavior:**
Currently, hovering over bars in the chart shows a tooltip with the name of the corresponding language and the number of its repos.

**Suggested behavior:**
Can we make the bars clickable?  Whenever a bar is clicked it opens up the user's profile filtering only the repos for that specific language.

For instance, if I click on the JavaScript bar in my [profile] I would be redirected to my profile filtering only the JavaScript [repos] I have.

The URL structure should be something like this:
```js
`https://github.com/${username}?tab=repositories&type=&language=${language}`
```

[profile]: https://githublangs.herokuapp.com/?username=ioayman
[repos]: https://github.com/IOAyman?tab=repositories&type=&language=javascript

### PR DESCRIPTION
**Pull request by masquel**
Open github repositories on bar click according to #39 issue

Changes: 
- added username to dataset of canvas-container to select it in chart.js
- added click event on chart bar to open github repos with selected language
- refactor DOM elements selecting

### CODE REVIEW

**Comment by wildan3105** [comment]
Thanks for the PR @masquel ! I'll review and test it first :ok_hand:

File: lib/public/js/chart.js [hunk 1]
```diff
@@ -73,6 +76,17 @@ const chart = ({ limit, singleColor }) => {
...
         }
     });
+    const onCanvasClick = (e) => {
+        const activeElement = chart.getElementAtEvent(e)[0];
+        if(activeElement){
+            const language = chart.data.labels[activeElement._index];
+            const value = chart.data.datasets[activeElement._datasetIndex].data[activeElement._index];
+            const url = `https://github.com/${username}?tab=repositories&type=&language=${language}`;
+            window.open(url, "_blank");
+            console.log(label, value);
```
Comment (wildan3105) [inline lib/public/js/chart.js]: please remove the `console.log`

**Comment by masquel** [comment]
Damn console.log. Done)

**Comment by wildan3105** [comment]
Okay I'll try it first on my local dev. Did you also try the `screenshot testing`? @masquel It's a new unit testing for this project to make sure every feature addition / bug patch would not destroy previous and in-use feature

**Comment by masquel** [comment]
@wildan3105 I've tried screenshot testing. Looks like I have different font type
homepage_0 fail
profile--non_existent_user_1 fail
profile--user_with_0_repos_2 fail
profile--user_with_more_than_0_repos_3 fail

**Comment by wildan3105** [comment]
Hi @masquel I have tested and it worked! But I wonder if we could add title below the language stats with something like : `click to browse the repos with the {language-name} programming language by ${username}`...
That would be wonderful!

**Comment by masquel** [comment]
@wildan3105, do you mean this tooltip should be instead of current one with statistic?

**Comment by wildan3105** [comment]
Yes tooltip! But I mean statistics **plus** `click to browse the repos with the {language-name} programming language by ${username}`, do you get that? @masquel, so maybe below the statistics text.
Or should we just add legend? This is your choice :smile: The point is user know that click the bar will eventually go to the repo with that language :ok_hand:

**Comment by masquel** [comment]
@wildan3105 language name is already in tooltip, so I've just added text `click to browse the repos with the current programming language by ${username}`. But it's don't look very good. Need more styles to become pretty)

**Comment by wildan3105** [comment]
Could you share the latest result (tooltip) of your change here please? @masquel

**Comment by masquel** [comment]
@wildan3105

**Comment by wildan3105** [comment]
Thanks! I'll take care of that :)

**Pull request merged**